source/agora/cli/checkvtable/main.d

```diff
@@ -0,0 +1,222 @@
+        The path to the file it creates is `source/scpp/extra/DVMChecks.cpp`.
+
+        The use of this is as follows.
+        agora-checkvtable source/scpp/extra/DVMChecks.cpp
+
+    Copyright:
+        Copyright (c) 2020 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module agora.cli.checkvtable.main;
+
+import scpd.tests.VTableTypes;
+import scpd.tests.VTableTest;
+
+import std.file;
+import std.format;
+import std.stdio;
+
+/// Workaround for issue likely related to dub #225,
+/// expects a main() function and invokes it after unittesting.
+version (unittest) void main () { } else:
+
+/*******************************************************************************
+
+    Application entry point
+
+*******************************************************************************/
+
+int main (string[] args)
+{
+    // Validate parameters
+    if (args.length < 2)
+    {
+        writeln("usage: agora-gen [output source file]");
+        return 1;
+    }
+    auto filename = args[1];
+    File file = File(filename, "w+");
+
+    bool first;
+
+    writeHeader(file);
+
+    static foreach (e; VTableCheckClasses)
+    {
+        mixin(
+        q{
+
+            writeMethodHeader(file, `%1$s`, `getVMOffset%1$s`);
+
+            first = true;
+            static foreach (idx, member; __traits(allMembers, %1$s))
+            {
+                mixin(
+                q{
+                    static if (__traits(isVirtualMethod, %1$s.%2$s) && (`%2$s` != `__dtor`) && (`%2$s` != `__xdtor`))
+                    {
+                        writeMethodContents(file, first, `%1$s`, `%2$s`);
+                        first = false;
+                    }
+                }.format(`%1$s`, member));
+            }
+
+            writeMethodFooter(file);
+
+            writeCheckFunction(file, `%1$s`);
+
+            file.writeln(``);
+
+        }.format(e, `%2$s`));
+    }
+
+
+    file.writeln(`int checkVMOffset (const char* classname, const char* offsets)`);
+    file.writeln(`{`);
+
+    static foreach (idx, e; VTableCheckClasses)
+    {
+        mixin(
+        q{
+            if (idx == 0)
+                file.writeln(`    if (strcmp(classname, "%1$s") == 0)`);
+            else
+                file.writeln(`    else if (strcmp(classname, "%1$s") == 0)`);
+            file.writeln(`    {`);
+            file.writeln(`        return checkVMOffset%1$s(offsets);`);
+            file.writeln(`    }`);
+        }.format(e));
+    }
+
+    file.writeln(`    else`);
+    file.writeln(`    {`);
+    file.writeln(`        return -1;`);
+    file.writeln(`    }`);
+    file.writeln(`}`);
+
+    file.close();
+
+    writefln("Source code was generated with the file name %s.", filename);
+
+    return 0;
+}
+
+/// Writes the header of the source file
+void writeHeader(File file)
```

Comment:
It should write to a buffer or an output sink (just like the Formatter and the Serializer)

Reply:
Also I have the same comment for all the other functions which write to a file.